src/System.Net.Http/tests/UnitTests/HPack/HuffmanDecodingTests.cs

```diff
@@ -3,31 +3,16 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Reflection;
 using Xunit;
+using System.Net.Http.HPack;
 
-namespace System.Net.Http.Functional.Tests
+namespace System.Net.Http.Unit.Tests.HPack
```

Comment:
In general we've tried to avoid adding new Unit tests that rely on calling directly into private product code. That's why these tests were originally written using reflection.

I don't have a strong opinion on this, but is there a reason these were changed?

@stephentoub @davidsh may have opinions here.

Reply:
>In general we've tried to avoid adding new Unit tests that rely on calling directly into private product code. That's why these tests were originally written using reflection.
I don't have a strong opinion on this, but is there a reason these were changed?

I have strong opinions here. :)

I prefer the UnitTests approach of including product code in test assemblies where it adds value especially for hard to test components that don't have easy refactoring for dependency injection.  That is approach we (System.Net/NCL) have always used in the past 5+ years. It follows the principles that we adopted for our "unit testing" approach including our naming convention based on this book:

https://smile.amazon.com/Art-Unit-Testing-examples/dp/1617290890/ref=sr_1_1?keywords=art+of+unit+testing&qid=1560986684&s=gateway&sr=8-1

Reply:
@geoffkizer Our functional tests do not include product code, but our unit tests do -- so the change here is simply to move it over from functional -> unit and get rid of the reflection. Sounds like @davidsh is aligned.